testsrc/main/mondrian/rolap/agg/AggregationOnInvalidRoleTest.java

```diff
@@ -0,0 +1,146 @@
+
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
+
+    @Override
+    protected TestContext createTestContext() {
+        // don't do anything dangerous here, just get standard context
+        return TestContext.instance();
+    }
+
+    private TestContext createContext() {
+        String cube = ""
+            + "<Cube name=\"mondrian2225\" visible=\"true\" cache=\"true\" enabled=\"true\">"
+            + "  <Table name=\"mondrian2225_fact\">"
+            + "    <AggName name=\"mondrian2225_agg\" ignorecase=\"true\">"
+            + "      <AggFactCount column=\"fact_count\"/>"
+            + "      <AggMeasure column=\"fact_Measure\" name=\"[Measures].[Measure]\"/>"
+            + "      <AggLevel column=\"dim_Code\" name=\"[Product Code].[Code]\" collapsed=\"true\"/>"
+            + "    </AggName>"
+            + "  </Table>"
+            + "  <Dimension type=\"StandardDimension\" visible=\"true\" foreignKey=\"customer_id\" highCardinality=\"false\" name=\"Customer\">"
+            + "    <Hierarchy name=\"Customer\" visible=\"true\" hasAll=\"true\" primaryKey=\"customer_id\">"
+            + "      <Table name=\"mondrian2225_customer\"/>"
+            + "        <Level name=\"First Name\" visible=\"true\" column=\"customer_name\" type=\"String\" uniqueMembers=\"false\" levelType=\"Regular\" hideMemberIf=\"Never\"/>"
+            + "    </Hierarchy>"
+            + "  </Dimension>"
+            + "  <Dimension type=\"StandardDimension\" visible=\"true\" foreignKey=\"product_ID\" highCardinality=\"false\" name=\"Product Code\">"
+            + "    <Hierarchy name=\"Product Code\" visible=\"true\" hasAll=\"true\" primaryKey=\"product_id\">"
+            + "      <Table name=\"mondrian2225_dim\"/>"
+            + "      <Level name=\"Code\" visible=\"true\" column=\"product_code\" type=\"String\" uniqueMembers=\"false\" levelType=\"Regular\" hideMemberIf=\"Never\"/>"
+            + "    </Hierarchy>"
+            + "  </Dimension>"
+            + "  <Measure name=\"Measure\" column=\"fact\" aggregator=\"sum\" visible=\"true\"/>"
+            + "</Cube>";
+
+        String role = ""
+            + "<Role name=\"Test\">"
+            + "  <SchemaGrant access=\"none\">"
+            + "    <CubeGrant cube=\"mondrian2225\" access=\"all\">"
+            + "      <HierarchyGrant hierarchy=\"[Customer.Customer]\" topLevel=\"[Customer.Customer].[First Name]\" access=\"custom\">"
+            + "        <MemberGrant member=\"[Customer.Customer].[NonExistingName]\" access=\"all\"/>"
+            + "      </HierarchyGrant>"
+            + "    </CubeGrant>"
+            + "  </SchemaGrant>"
+            + "</Role>";
+
+        return TestContext.instance()
+            .create(null, cube, null, null, null, role)
+            .withRole("Test");
+    }
+
+
+    public void test_ThrowsException_WhenNonIgnoringInvalidMembers() {
+        propSaver.set(propSaver.properties.IgnoreInvalidMembers, false);
```

Comment:
This test is order dependent.  If the other test runs first then the schema will be stored in the pool and reused.  I happened to see this fail when running it myself because the other test ran first.
Flushing the cache at the start of this test might be sufficient to address that.